### PR TITLE
unify Win32 macro to __MINGW32__

### PR DESCRIPTION
**Pull request by cielavenir**
We should use only one macro for such purpose

### CODE REVIEW

**Comment by cielavenir** [comment]
@gaborcsardi now fixing archive_write_files on Windows depends on this PR. Please take a look.

**Comment by cielavenir** [comment]
do you have any comments on this?

File: src/archive_write_files.cpp [hunk 1]
```diff
@@ -48,7 +48,7 @@ const char* const pb_format =
   for (std::string file : files) {
     stat(file.c_str(), &st);
     entry = archive_entry_new();
-#if defined(_WIN32) || (!defined(__GNUC__) && !defined(__clang__))
```
Comment (gaborcsardi) [inline src/archive_write_files.cpp:51]: I am not sure why we need this PR to be honest. Especially this line, which also changes behavior on non-GNU compilers, not only on Windows.

Comment (cielavenir) [reply src/archive_write_files.cpp:51]: then what about https://github.com/r-lib/archive/pull/80#discussion_r1234284267 ... your stance is different...

**Pull request closed without merging**